vpr/src/place/place_checkpoint.cpp

```diff
@@ -43,6 +44,11 @@ void restore_best_placement(t_placement_checkpoint& placement_checkpoint, std::s
                                    timing_info.get(),
                                    &costs);
 
+        // If NoC is enabled, re-compute NoC costs and re-initialize NoC internal data structures
```

Comment:
Good to say why (the NoC placement may have changed, and hence the routing of traffic flows and NoC costs need to be recomputed).